test/test_subdomains.py

```diff
@@ -40,9 +41,37 @@ def test_different_volume_ids():
     vol_subdomain_3 = F.VolumeSubdomain1D(
         id=vol_subdomains_ids[2], borders=[sub_dom_3[0], sub_dom_3[-1]], material=None
     )
-    my_model.subdomains = [vol_subdomain_1, vol_subdomain_2, vol_subdomain_3]
+    my_model.subdomains = [vol_subdomain_1, vol_subdomain_3, vol_subdomain_2]
 
     my_model.define_markers_and_measures()
 
     for vol_id in vol_subdomains_ids:
         assert vol_id in np.array(my_model.volume_meshtags.values)
+
+
+def test_non_matching_volume_borders():
+    my_model = F.HydrogenTransportProblem()
+
+    sub_dom_1 = np.linspace(0, 1e-04, num=5)
+    sub_dom_2 = np.linspace(8e-05, 2e-04, num=5)
+    my_model.mesh = F.Mesh1D(np.unique(np.concatenate([sub_dom_1, sub_dom_2])))
+
+    vol_subdomains_ids = [1, 2]
+    vol_subdomain_1 = F.VolumeSubdomain1D(
+        id=vol_subdomains_ids[0], borders=[sub_dom_1[0], sub_dom_1[-1]], material=None
+    )
+    vol_subdomain_2 = F.VolumeSubdomain1D(
+        id=vol_subdomains_ids[1], borders=[sub_dom_2[0], sub_dom_2[-1]], material=None
+    )
```

Comment:
This can be simplified.

```suggestion
    my_model.mesh = F.Mesh1D(np.linspace(0, 10))

    vol_subdomain_1 = F.VolumeSubdomain1D(
        id=1, borders=[0, 2], material=None
    )
    vol_subdomain_2 = F.VolumeSubdomain1D(
        id=2, borders=[4, 10], material=None
    )
```